### DISCUSSION TITLE
Force loading as linear RGB

### DISCUSSION THREAD

**Discussion by tuket** [KhronosGroup/KTX-Software#488] [Q&A] [answered]
Hi,

I have a ktx2 texture that is marked as sRGB. However, I would like to upload it to OpenGL as linear RGB.

In ktx1 I was able to do it by modifying the `glInternalFormat`.

```cpp
typedef struct ktxTexture1 {
    KTXTEXTURECLASSDEFN
    ktx_uint32_t glFormat; /*!< Format of the texture data, e.g., GL_RGB. */
    ktx_uint32_t glInternalformat; /*!< Internal format of the texture data,
                                        e.g., GL_RGB8. */
    ktx_uint32_t glBaseInternalformat; /*!< Base format of the texture data,
                                            e.g., GL_RGB. */
    ktx_uint32_t glType; /*!< Type of the texture data, e.g, GL_UNSIGNED_BYTE.*/
    struct ktxTexture1_private* _private; /*!< Private data. */
} ktxTexture1;
```

But now, with ktx2, I down't know how to do it.

```cpp
typedef struct ktxTexture2 {
    KTXTEXTURECLASSDEFN
    ktx_uint32_t  vkFormat;
    ktx_uint32_t* pDfd;
    ktxSupercmpScheme supercompressionScheme;
    ktx_bool_t isVideo;
    ktx_uint32_t duration;
    ktx_uint32_t timescale;
    ktx_uint32_t loopcount;
    struct ktxTexture2_private* _private;  /*!< Private data. */
} ktxTexture2;
```

Cheers.

**Comment by MarkCallow** [comment, marked as answer]
You will have to modify the vkFormat. Looking at the enum names it is easy to see what to change. Working with the numeric values it is harder. If you look up the vkFormat value in the Vulkan VkFormat enum list you will find its name has `SRGB` in it. The matching non-sRGB format will have `UNORM` in place of  `SRGB` in its name.

Why do you want to do this? Is the texture incorrectly marked as sRGB?

**Reply by tuket** [reply]
Yes, the texture is incorrectly marked as sRGB.
Okay, I will do that.
Thanks for your response!

**Reply by MarkCallow** [reply]
> Yes, the texture is incorrectly marked as sRGB.

I suggest using a binary editor, e.g. iHex on Mac, to fix the file. Maybe we should add an option to `ktx2ktx2` to change the format. Would you like to provide a PR?
